pkg/segment/query/segquery.go

```diff
@@ -813,6 +813,13 @@ func applyAggOpOnSegments(sortedQSRSlice []*QuerySegmentRequest, allSegFileResul
 		}
 		isSegmentFullyEnclosed := segReq.segKeyTsRange.AreTimesFullyEnclosed(segReq.segKeyTsRange.StartEpochMs, segReq.segKeyTsRange.EndEpochMs)
 
+		// For Unrotated search, Check if the segment is rotated and update the search type accordingly
+		if segReq.sType != structs.SEGMENT_STATS_SEARCH {
+			if utils.IsFileForRotatedSegment(segReq.segKey) {
+				segReq.sType = structs.SEGMENT_STATS_SEARCH
+			}
+		}
```

Comment:
nit: I think it would be a little more clear if you change `if segReq.sType != structs.SEGMENT_STATS_SEARCH` to `if segReq.sType = structs.UNROTATED_SEGMENT_STATS_SEARCH`, since I think that's the only case you want to handle. Or are there also other values that would be valid at this point in the flow?